flamingpy/utils/viz.py

```diff
@@ -205,53 +197,132 @@ def draw_EGraph(
+                the tuple should be of the form:
+                ``("weight",{-1: minus_color, +1: plus_color})``.
+
+    Returns:
+        A Matplotib Axes object.
+    """
+    # Plotting edges.
+    for edge in egraph.edges:
+        # Color edges based on `color_eges`:
+        #  if `color_edges` is a string use it as color,
+        # using the attribute and color dict if `color_edges` is a tuple(str,dict)
+        # or based on color attribute (when available) if `color_edges` is bool and True;
+        # black otherwise.
+        if isinstance(color_edges, str):
+            color = color_edges
+        elif isinstance(color_edges, tuple):
+            edge_attribute, color_dict = color_edges
+            if not (isinstance(edge_attribute, str) and isinstance(color_dict, dict)):
+                raise ValueError(
+                    "Inappropiate value for `color_edges` argument:"
+                    "Check it complies with the type `tuple(str, dict)`"
+                    "where the string corresponds to a valid edge attribute,"
+                    "the dictionary keys to valid attribute values and"
+                    "dictionary values to valid matplotlib color strings."
+                )
+            edge_property = egraph.edges[edge].get(edge_attribute)
+            color = color_dict.get(edge_property)
+        elif isinstance(color_edges, bool) and color_edges == True:
+            color = egraph.edges[edge].get("color") if color_edges else "grey"
+        else:
+            color = "k"
+
+        x1, z1, y1 = edge[0]
+        x2, z2, y2 = edge[1]
+        ax.plot([x1, x2], [y1, y2], [z1, z2], color=color, linewidth=0.5)
+
+    return ax
+
+
+def _plot_EGraph_nodes(ax, egraph, color_nodes, label, name, legend):
+    """Draw the nodes of the graph state represented by the EGraph.
+
+    Args:
+        ax (matplotlib.axes.Axes): the axes to draw the points in
+        color_nodes (bool or string or dict): Options are:
+
+            True: color the nodes based on the 'color' attribute
+                attached to the node. If unavailable, color nodes black.
+            string: color all nodes with the color specified by the string
+            tuple[str, dict]: color nodes based on attribute and defined colour
+                string by providing a tuple with [attribute, color_dictionary],
+                for example: ``["state", {"GKP": "b", "p": "r"}]``
+                will look at the "state" attribute of the node, and colour
+                according to the dictionary.
+
+        label (NoneType or string): plot values next to each node
+            associated with the node attribute label. For example,
+            to plot bit values, set label to "bit_val". If set to 'index',
+            it will plot the integer indices of the nodes. If the attribute
+            for some or all of the nodes, a message will print indicating
+            for how many nodes the attribute has not been set.
```

Comment:
```suggestion
        label (NoneType or string): plot values next to each node
            associated with the node attribute label. For example,
            to plot bit values, set label to "bit_val". If set to 'index',
            it will plot the integer indices of the nodes.
```